Change all unary functions stubs to use TensorIteratorBase&

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56151 Port all non-float unary operators to structured (and rsqrt)
* #56082 Port all unary float functions to structured
* **#56078 Change all unary functions stubs to use TensorIteratorBase&**

This is in preparation for making all unary functions structured.
I don't actually have to make them structured yet as TensorIterator&
casts to TensorIteratorBase&

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D27777768](https://our.internmc.facebook.com/intern/diff/D27777768)

ezyang committed:
Change all unary functions stubs to use TensorIteratorBase&

This is in preparation for making all unary functions structured.
I don't actually have to make them structured yet as TensorIterator&
casts to TensorIteratorBase&

Distributions functions seem to live in other files so I left them
alone for now.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Change all unary functions stubs to use TensorIteratorBase&"


This is in preparation for making all unary functions structured.
I don't actually have to make them structured yet as TensorIterator&
casts to TensorIteratorBase&

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

SplitInfinity commented on aten/src/ATen/native/TensorFactories.cpp:
```diff
@@ -14,6 +14,7 @@
 #include <ATen/detail/CUDAHooksInterface.h>
 #include <c10/util/Exception.h>
 #include <ATen/NamedTensorUtils.h>
+#include <ATen/native/UnaryOps.h>
```
Are there any uses of this header in this file?

ezyang replied:
yeah, I moved `kaiser_window_stub` into UnaryOps.h because it's defined in one of the CUDA unary ops files, and then I had to fixup it here. This one probably shouldn't be using stub though.

ezyang committed:
Update on "Change all unary functions stubs to use TensorIteratorBase&"


This is in preparation for making all unary functions structured.
I don't actually have to make them structured yet as TensorIterator&
casts to TensorIteratorBase&

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D27777768](https://our.internmc.facebook.com/intern/diff/D27777768)

[ghstack-poisoned]

ezyang committed:
Update on "Change all unary functions stubs to use TensorIteratorBase&"


This is in preparation for making all unary functions structured.
I don't actually have to make them structured yet as TensorIterator&
casts to TensorIteratorBase&

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D27777768](https://our.internmc.facebook.com/intern/diff/D27777768)

[ghstack-poisoned]

ezyang committed:
Update on "Change all unary functions stubs to use TensorIteratorBase&"


This is in preparation for making all unary functions structured.
I don't actually have to make them structured yet as TensorIterator&
casts to TensorIteratorBase&

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D27777768](https://our.internmc.facebook.com/intern/diff/D27777768)

[ghstack-poisoned]
